_docs/src/documentation/test_projects.md

```diff
@@ -0,0 +1,348 @@
+# How To Create Test Projects
+
+## Type Of Test Project
+
+Rex supports 3 different kind of test projects, each requiring a slightly different setup.
+
+### Unit Test Project
+A unit test project is ideal to test a library or system that doesn't need to engine to be in a specific state. If you have a container, allocator, algorithm, .. that you'd like to test, a unit test is ideal to do this.
+
+Let's work from an example. Let's pretend we want to create a unit test project for the rex engine project called `RexUnitTests`.
```

Comment:
It would be a nice feature if there would be an automated step to setup these projects. for instance within VS they can just do File > New Project > Rex Unit Test Template and all the code is generated and put in place. It might also be a command line argument like we already do for enums and array we could also support generation of projects.